### ISSUE TITLE
how to build with RHEL 6

### ISSUE THREAD

**Issue by yuusuke** [ClusterLabs/resource-agents#1292]
We packaged resource-agents for RHEL 6.
As of v4.2.0 and later, since the python library was added, the build will now fail with RHEL 6.
What should we do? Is there an option to exclude python library from build target? Or are you planning to add it?

**Comment by oalbrigt** [comment]
You can use `./configure --with-agents=` to specify which agents to run checks for/build.

**Comment by yuusuke** [comment]
Thanks for comment.
I tried specifying IPaddr 2 but it failed.
Output of "configure: WARNING: unrecognized options: --with-agents" was output in configure log, is this option now available?

```
[root@42f7472df406 resource-agents]# ./autogen.sh
autoreconf: Entering directory `.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal
autoreconf: configure.ac: tracing
autoreconf: configure.ac: not using Libtool
autoreconf: running: /usr/bin/autoconf
autoreconf: running: /usr/bin/autoheader
autoreconf: running: automake --add-missing --copy --no-force
heartbeat/Makefile.am:53: whitespace following trailing backslash
autoreconf: Leaving directory `.'
Now run ./configure and make
[root@42f7472df406 resource-agents]# ./configure --with-ras-set=linux-ha --with-compat-habindir --with-agents=IPaddr2
configure: WARNING: unrecognized options: --with-agents
(snip)
resource-agents configuration:
  Version                  = 4.2.0
  Build Version            = :%H$
  Features                 =

  Prefix                   = /usr
  Executables              = /usr/sbin
  Man pages                = /usr/share/man
  Libraries                = /usr/lib64
  Header files             = /usr/include
  Arch-independent files   = /usr/share
  Documentation            = /usr/share/doc/resource-agents
  State information        = /var
  System configuration     = /etc
  HA_BIN directory prefix  = /usr/lib64
  RA state files           = /var/run/resource-agents
  AIS Plugins              =

  CFLAGS                   = -g -O2 -ggdb3  -fgnu89-inline -fstack-protector-all -Wall -Wbad-function-cast -Wcast-qual -Wdeclaration-after-statement -Wendif-labels -Wfloat-equal -Wformat=2 -Wformat-security -Wformat-nonliteral -Winline -Wmissing-prototypes -Wmissing-declarations -Wmissing-format-attribute -Wnested-externs -Wno-long-long -Wno-strict-aliasing -Wpointer-arith -Wstrict-prototypes -Wwrite-strings -ansi -D_GNU_SOURCE -DANSI_ONLY -Werror
  Libraries                =  -lglib-2.0
  Stack Libraries          =
[root@42f7472df406 resource-agents]# make
(snip)
gmake[3]: Entering directory `/usr/local/src/repo/resource-agents/doc/man'
OCF_ROOT=. OCF_FUNCTIONS_DIR=../../heartbeat ../../heartbeat/azure-events meta-data > metadata-azure-events.xml
Traceback (most recent call last):
  File "../../heartbeat/azure-events", line 823, in <module>
    main()
  File "../../heartbeat/azure-events", line 819, in main
    setLoglevel(ocf.is_true(ocf.get_parameter("verbose", "false")))
  File "/usr/local/src/repo/resource-agents/heartbeat/ocf.py", line 206, in get_parameter
    return env.get("OCF_RESKEY_{}".format(name), default)
ValueError: zero length field name in format
gmake[3]: *** [metadata-azure-events.xml] Error 1
rm metadata-azure-events.xml
gmake[3]: Leaving directory `/usr/local/src/repo/resource-agents/doc/man'
gmake[2]: *** [all-recursive] Error 1
gmake[2]: Leaving directory `/usr/local/src/repo/resource-agents/doc'
gmake[1]: *** [all-recursive] Error 1
gmake[1]: Leaving directory `/usr/local/src/repo/resource-agents'
make: *** [all] Error 2
```

**Comment by oalbrigt** [comment]
This patch adds auto-detection, so if you apply it it will skip the agents you havent got libraries for:
https://github.com/ClusterLabs/resource-agents/pull/1294

**Comment by yuusuke** [comment]
Thank you for making patches!
However, although I tried it, I got an error at the stage of autogen.sh.
Do I need to do something?

```
[root@51289712ebe1 opt]# git clone https://github.com/ClusterLabs/resource-agents.git
Initialized empty Git repository in /opt/resource-agents/.git/
remote: Enumerating objects: 12, done.
remote: Counting objects: 100% (12/12), done.
remote: Compressing objects: 100% (12/12), done.
remote: Total 59591 (delta 5), reused 0 (delta 0), pack-reused 59579
Receiving objects: 100% (59591/59591), 31.23 MiB | 3.79 MiB/s, done.
Resolving deltas: 100% (39389/39389), done.
[root@51289712ebe1 opt]# cd resource-agents/
[root@51289712ebe1 resource-agents]# git checkout v4.2.0
Note: checking out 'v4.2.0'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b new_branch_name

HEAD is now at b222ea8... build: update ChangeLog for 4.2.0
[root@51289712ebe1 resource-agents]# curl -LO https://github.com/ClusterLabs/resource-agents/pull/1294.patch
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
101  6390    0  6390    0     0   4599      0 --:--:--  0:00:01 --:--:-- 22821
[root@51289712ebe1 resource-agents]# patch -d. -p1 <1294.patch
patching file .gitignore
patching file configure.ac
patching file doc/man/Makefile.am
Hunk #3 succeeded at 177 (offset -1 lines).
patching file heartbeat/Makefile.am
Hunk #3 succeeded at 168 (offset -1 lines).
patching file m4/ac_python_module.m4
[root@51289712ebe1 resource-agents]# ./autogen.sh
autoreconf: Entering directory `.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal
autoreconf: configure.ac: tracing
autoreconf: configure.ac: not using Libtool
autoreconf: running: /usr/bin/autoconf
configure.ac:514: error: possibly undefined macro: AC_PYTHON_MODULE
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
autoreconf: /usr/bin/autoconf failed with exit status: 1
[root@51289712ebe1 resource-agents]#
```

**Comment by oalbrigt** [comment]
Try downloading the latest configure.ac: https://raw.githubusercontent.com/ClusterLabs/resource-agents/master/configure.ac

if that doesnt help, there might be some additional paramter that needs to be added to it for older autoconf versions.

**Comment by inouekazu** [comment]
There are two issues left.

 * my environment
   ```
   [root@27ccbf84dffc resource-agents]# cat /etc/redhat-release
   CentOS release 6.10 (Final)

   [root@27ccbf84dffc resource-agents]# git show
   commit 45b55fae82b9fcb3aa9107dd37e2c36e3335de01
   Merge: 7ee7136 6303448
   Author: Kristoffer Gronlund <krig@koru.se>
   Date:   Wed Feb 27 10:11:48 2019 +0100

       Merge pull request #1302 from mbaldessari/redis-warning

       redis: Filter warning from stderr when calling 'redis-cli -a'
   ```
1. AC_PYTHON_MODULE has not been load.
   ```
   [root@27ccbf84dffc resource-agents]# ./autogen.sh
   autoreconf: Entering directory `.'
   autoreconf: configure.ac: not using Gettext
   autoreconf: running: aclocal
   autoreconf: configure.ac: tracing
   autoreconf: configure.ac: not using Libtool
   autoreconf: running: /usr/bin/autoconf
   configure.ac:514: error: possibly undefined macro: AC_PYTHON_MODULE
         If this token and others are legitimate, please use m4_pattern_allow.
         See the Autoconf documentation.
   autoreconf: /usr/bin/autoconf failed with exit status: 1

   [root@27ccbf84dffc resource-agents]# tail aclocal.m4
     fi
   done
   rm -rf conftest.dir

   AC_CACHE_VAL([am_cv_prog_tar_$1], [am_cv_prog_tar_$1=$_am_tool])
   AC_MSG_RESULT([$am_cv_prog_tar_$1])])
   AC_SUBST([am__tar])
   AC_SUBST([am__untar])
   ]) # _AM_PROG_TAR

   [root@27ccbf84dffc resource-agents]#
   ```
   It's solved by specifying ACLOCAL_AMFLAGS in Makefile.am: https://github.com/inouekazu/resource-agents/commit/d5f56e3115f861dc7d69686c7fffe05bddc8a2c1
   ```
   [root@27ccbf84dffc resource-agents]# ./autogen.sh
   autoreconf: Entering directory `.'
   autoreconf: configure.ac: not using Gettext
   autoreconf: running: aclocal -I m4
   autoreconf: configure.ac: tracing
   autoreconf: configure.ac: not using Libtool
   autoreconf: running: /usr/bin/autoconf
   autoreconf: running: /usr/bin/autoheader
   autoreconf: running: automake --add-missing --copy --no-force
   configure.ac:112: installing `./compile'
   configure.ac:35: installing `./config.guess'
   configure.ac:35: installing `./config.sub'
   configure.ac:93: installing `./install-sh'
   configure.ac:93: installing `./missing'
   heartbeat/Makefile.am: installing `./depcomp'
   autoreconf: Leaving directory `.'
   Now run ./configure and make

   [root@27ccbf84dffc resource-agents]# tail aclocal.m4
   done
   rm -rf conftest.dir

   AC_CACHE_VAL([am_cv_prog_tar_$1], [am_cv_prog_tar_$1=$_am_tool])
   AC_MSG_RESULT([$am_cv_prog_tar_$1])])
   AC_SUBST([am__tar])
   AC_SUBST([am__untar])
   ]) # _AM_PROG_TAR

   m4_include([m4/ac_python_module.m4])
   [root@27ccbf84dffc resource-agents]#
   ```
2. I think that ```azure-events```, ```gcp-pd-move, etc.``` that use ```heartbeat/ocf.py``` need to skip building in environments **under Python 2.6 or less**.
   * [Resource Agent guide for Python](https://github.com/ClusterLabs/resource-agents/blob/v4.2.0/doc/dev-guides/writing-python-agents.md#introduction)
     > Agents written in Python should be ideally compatible both with Python
     > 2.7+ and Python 3.3+.

   ```
   [root@27ccbf84dffc resource-agents]# ./configure
    :
   checking for python... /usr/bin/python
   checking for python version... 2.6
   checking for python platform... linux2
   checking for python script directory... ${prefix}/lib/python2.6/site-packages
   checking for python extension module directory... ${exec_prefix}/lib64/python2.6/site-packages
   checking python module: googleapiclient... no
   checking python module: pyroute2... no
   configure: WARNING: "Not building gcp-pd-move"
   configure: WARNING: "Not building gcp-vpc-move-route"
   configure: WARNING: "Not building gcp-vpc-move-vip"
    :

   [root@27ccbf84dffc resource-agents]# make
    :
   OCF_ROOT=. OCF_FUNCTIONS_DIR=../../heartbeat ../../heartbeat/azure-events meta-data > metadata-azure-events.xml
   Traceback (most recent call last):
     File "../../heartbeat/azure-events", line 823, in <module>
       main()
     File "../../heartbeat/azure-events", line 819, in main
       setLoglevel(ocf.is_true(ocf.get_parameter("verbose", "false")))
     File "/usr/local/src/repo/resource-agents/heartbeat/ocf.py", line 206, in get_parameter
       return env.get("OCF_RESKEY_{}".format(name), default)
   ValueError: zero length field name in format
   gmake[3]: *** [metadata-azure-events.xml] Error 1
   rm metadata-redis.xml metadata-ovsmonitor.xml metadata-Pure-FTPd.xml metadata-fio.xml metadata-galera.xml metadata-mpathpersist.xml metadata-rkt.xml metadata-symlink.xml metadata-ipsec.xml metadata-eDir88.xml metadata-ManageRAID.xml metadata-oralsnr.xml metadata-openstack-cinder-volume.xml metadata-lxc.xml metadata-sfex.xml metadata-podman.xml metadata-varnish.xml metadata-Xen.xml metadata-iface-bridge.xml metadata-Xinetd.xml metadata-NodeUtilization.xml metadata-kamailio.xml metadata-iface-vlan.xml metadata-openstack-floating-ip.xml metadata-ICP.xml metadata-sybaseASE.xml metadata-garbd.xml metadata-Squid.xml metadata-IPaddr.xml metadata-IPsrcaddr.xml metadata-proftpd.xml metadata-lxd-info.xml metadata-VirtualDomain.xml metadata-EvmsSCC.xml metadata-jboss.xml metadata-WAS.xml metadata-docker.xml metadata-MailTo.xml metadata-Evmsd.xml metadata-vsftpd.xml metadata-iSCSILogicalUnit.xml metadata-slapd.xml metadata-syslog-ng.xml metadata-openstack-info.xml metadata-ClusterMon.xml metadata-   Filesystem.xml metadata-SphinxSearchDaemon.xml metadata-rabbitmq-cluster.xml metadata-asterisk.xml metadata-iscsi.xml metadata-ManageVE.xml metadata-Stateful.xml metadata-AudibleAlarm.xml metadata-machine-info.xml metadata-oraasm.xml metadata-conntrackd.xml metadata-dnsupdate.xml metadata-rsyncd.xml metadata-vmware.xml metadata-SAPDatabase.xml metadata-jira.xml metadata-IPaddr2.xml metadata-Raid1.xml metadata-nagios.xml metadata-aws-vpc-move-ip.xml metadata-iSCSITarget.xml metadata-LVM.xml metadata-nfsserver.xml metadata-oracle.xml metadata-ids.xml metadata-azure-events.xml metadata-nfsnotify.xml metadata-CTDB.xml metadata-nginx.xml metadata-ethmonitor.xml metadata-pgsql.xml metadata-LVM-activate.xml metadata-Route.xml metadata-clvm.xml metadata-apache.xml metadata-ZFS.xml metadata-named.xml metadata-postfix.xml metadata-SAPInstance.xml metadata-SendArp.xml metadata-AoEtarget.xml metadata-tomcat.xml metadata-SysInfo.xml metadata-lvmlockd.xml metadata-pound.xml metadata-Dummy.xml    metadata-awseip.xml metadata-scsi2reservation.xml metadata-azure-lb.xml metadata-dhcpd.xml metadata-zabbixserver.xml metadata-rsyslog.xml metadata-ServeRAID.xml metadata-aliyun-vpc-move-ip.xml metadata-pingd.xml metadata-exportfs.xml metadata-awsvip.xml metadata-WinPopup.xml metadata-vdo-vol.xml metadata-db2.xml metadata-VIPArip.xml metadata-WAS6.xml metadata-minio.xml metadata-Delay.xml metadata-gcp-vpc-move-ip.xml metadata-portblock.xml metadata-LinuxSCSI.xml metadata-anything.xml metadata-aws-vpc-route53.xml metadata-mysql-proxy.xml metadata-pgagent.xml metadata-mysql.xml metadata-mariadb.xml metadata-sg_persist.xml
   gmake[3]: Leaving directory `/usr/local/src/repo/resource-agents/doc/man'
   gmake[2]: *** [all-recursive] Error 1
   gmake[2]: Leaving directory `/usr/local/src/repo/resource-agents/doc'
   gmake[1]: *** [all-recursive] Error 1
   gmake[1]: Leaving directory `/usr/local/src/repo/resource-agents'
   make: *** [all] Error 2
   [root@27ccbf84dffc resource-agents]#
   ```
   I tried to make a patch: https://github.com/inouekazu/resource-agents/commit/09f7f8b134648fb4b239853f81479c3b48f9673e
   ```
   [root@27ccbf84dffc resource-agents]# ./configure
    :
   checking for python... /usr/bin/python
   checking for python version... 2.6
   checking for python platform... linux2
   checking for python script directory... ${prefix}/lib/python2.6/site-packages
   checking for python extension module directory... ${exec_prefix}/lib64/python2.6/site-packages
   checking python module: googleapiclient... no
   checking python module: pyroute2... no
   configure: WARNING: "Not building azure-events"
   configure: WARNING: "Not building gcp-pd-move"
   configure: WARNING: "Not building gcp-vpc-move-route"
   configure: WARNING: "Not building gcp-vpc-move-vip"
    :
   [root@27ccbf84dffc resource-agents]#
   ```

**Comment by yuusuke** [comment]
This issue has been closed to #1308 so we will close it.
